test/langtools/tools/javac/plugin/AutostartPlugins.java

```diff
@@ -30,7 +30,7 @@
  *           jdk.compiler/com.sun.tools.javac.main
  *           jdk.jlink
  *  @build toolbox.ToolBox toolbox.JavacTask toolbox.JarTask
- *  @run main AutostartPlugins
+ *  @run main/othervm AutostartPlugins
```

Comment:
Why do the two javac plugin tests changed to /othervm? I assume they fail with a JDK built with --generate-linkable-runtime but I can't immediately see why.

Reply:
jtreg patches the `java.base` module and those langtools tests *link* from the run-time image (in such a config). Having a patched JDK module is not supported (and I don't think we ever should). Running in a separate JVM doesn't have this problem. See https://github.com/openjdk/jdk/pull/14787/commits/d8e1e834508589725adb8d70acb862a1270678ca. Does that make sense?

Reply:
That's okay, I wasn't initially sure why they were changed.  I'm looking at JRTArchiveFile.toEntry and wondering there should be a follow-up issue (not this PR) to fail early if running on a patched run-time even though it would be an odd configuration to attempt to do that.

Reply:
There already is. See:
https://github.com/openjdk/jdk/pull/14787/files#diff-b6b47eacb6060eb0a583a253f322f5d274063e082a12a72e8628a6e1ba6cdd3eR466-R471

It's also tested with [PatchedJDKModuleJlinkTest.java](https://github.com/openjdk/jdk/pull/14787/files#diff-11b8a26307346fd7ca016a349243cabd3b982964aaf4335298e28e956b3968eb). Do we need more?

Reply:
Yes, I know there is a test. It's "no such file" case in JRTArchiveFile.toEntry that I'm looking at.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8343839 for this.